Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -11717,6 +11717,36 @@ void SpeculativeJIT::compileNumberToStringWithValidRadixConstant(Node* node, int
 
     switch (node->child1().useKind()) {
     case Int32Use: {
+        if (radix == 10) {
+            SpeculateStrictInt32Operand value(this, node->child1());
+            GPRTemporary result(this);
+
+            GPRReg valueGPR = value.gpr();
+            GPRReg resultGPR = result.gpr();
+
+            flushRegisters();
+
+            JumpList slowCases;
+            JumpList doneCases;
+
+            slowCases.append(branch32(AboveOrEqual, valueGPR, TrustedImm32(NumericStrings::cacheSize)));
+
+            move(valueGPR, resultGPR);
+            lshiftPtr(Imm32(sizeof(NumericStrings::StringWithJSString) == 16 ? 4 : 3), resultGPR);
```

Comment:
Use `TrustedImm32` because This immediate is not dependent on remote code.

nit: maybe we can make `fastLog2()` `constexpr` in MathExtras.h and use that here instead of doing this conditional statement to compute the shift?  Also `static_assert` that `1 << shiftValue` produces `sizeof(NumericStrings::StringWithJSString)`?

Reply:
Changed.